lib/transaction.js

```diff
@@ -81,12 +98,16 @@ Transaction.prototype.commit = function() {
     .sequelize
     .getQueryInterface()
     .commitTransaction(this, this.options)
-    .finally(function() {
+    .finally(function () {
       self.finished = 'commit';
       if (!self.options.transaction) {
         self.cleanup();
       }
-    });
+    })
+    .tap(
+      function (val) { self.promise.seqResolve(val); },
+      function (err) { self.promise.seqReject(err); }
```

Comment:
bluebird specifically prevents that https://github.com/petkaantonov/bluebird/blob/master/src/promise.js#L45 and enforces you to use  composition instead. 

However, i do believe that with https://github.com/sequelize/sequelize/pull/3265 we can remove the custom promise constructor completely